### PR TITLE
executor: fix unstable TestGlobalMemoryTrackerOnCleanUp

### LINKED ISSUE #36585: unstable test in the TestGlobalMemoryTrackerOnCleanUp
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2022-07-26T09:44:03.874Z]     mem_test.go:46: 
[2022-07-26T09:44:03.874Z]         	Error Trace:	mem_test.go:46
[2022-07-26T09:44:03.874Z]         	Error:      	Not equal: 
[2022-07-26T09:44:03.874Z]         	            	expected: 2118
[2022-07-26T09:44:03.874Z]         	            	actual  : 0
[2022-07-26T09:44:03.874Z]         	Test:       	TestGlobalMemoryTrackerOnCleanUp
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl_workerpool.go:82] ["[ddl] closing workerPool"]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl_worker.go:165] ["[ddl] DDL worker closed"] [worker="worker 17, tp add index"] ["take time"=2.104µs]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl_worker.go:165] ["[ddl] DDL worker closed"] [worker="worker 18, tp add index"] ["take time"=1.485µs]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl_worker.go:165] ["[ddl] DDL worker closed"] [worker="worker 19, tp add index"] ["take time"=1.097µs]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl_workerpool.go:82] ["[ddl] closing workerPool"]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl_worker.go:165] ["[ddl] DDL worker closed"] [worker="worker 16, tp general"] ["take time"=1.207µs]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl_worker.go:165] ["[ddl] DDL worker closed"] [worker="worker 15, tp add index"] ["take time"=755ns]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl_worker.go:165] ["[ddl] DDL worker closed"] [worker="worker 14, tp general"] ["take time"=410ns]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [delete_range.go:148] ["[ddl] closing delRange"]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [session_pool.go:94] ["[ddl] closing sessionPool"]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl.go:731] ["[ddl] DDL closed"] [ID=44866c91-8b5b-42e0-9e3e-ff9d80bad69d] ["take time"=4.471582ms]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.859 +08:00] [INFO] [ddl.go:583] ["[ddl] stop DDL"] [ID=44866c91-8b5b-42e0-9e3e-ff9d80bad69d]
[2022-07-26T09:44:03.874Z] [2022/07/26 17:44:00.860 +08:00] [INFO] [domain.go:501] ["topNSlowQueryLoop exited."]
[2022-07-26T09:44:03.875Z] [2022/07/26 17:44:00.860 +08:00] [INFO] [handle_hist.go:156] ["SubLoadWorker exited."]
```

https://prow.pingcap.net/view/gs/pingcapprow/logs/bazel_test_tidb/1560501401715478528#1:build-log.txt%3A461

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close #36585

Problem Summary:

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/insert.go [hunk 1]
```diff
@@ -315,6 +315,9 @@ func (e *InsertExec) Next(ctx context.Context, req *chunk.Chunk) error {
 
 // Close implements the Executor Close interface.
 func (e *InsertExec) Close() error {
+	if e.memTracker != nil {
+		defer e.memTracker.Detach()
```
Comment (ekexium) [inline executor/insert.go:319]: Just curious: is it necessary?

Comment (Yisaer) [reply executor/insert.go]: updated.

Comment (ekexium) [reply executor/insert.go]: Is it possible to use more memory after the executor is closed? I didn't get it why we need to detach it.

**Review by ekexium — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**